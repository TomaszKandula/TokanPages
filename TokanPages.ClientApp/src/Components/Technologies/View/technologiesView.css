.technology-box {
    padding-top: 64px;
    padding-bottom: 64px;
}

.technology-icon {
    width: 32px !important;
    height: 32px !important;
    color: #6367ef !important;
    margin-right: 8px;
}

.technology-skeleton-circle {
    width: 42px !important;
    height: 42px !important;
    margin: 0 15px 0 0 !important;
}

.technology-caption-text {
    text-align: center;
    font-size: 2rem !important;
    font-weight: 500 !important;
    color: #6367ef;
}

.technology-feature-title {
    font-size: 1.5rem !important;
    font-weight: 400 !important;
    margin: 0;
    color: black;
}

.technology-feature-text {
    font-size: 1.125rem !important;
    font-weight: 400 !important;
    line-height: 1.8;
    color: #757575;
}
