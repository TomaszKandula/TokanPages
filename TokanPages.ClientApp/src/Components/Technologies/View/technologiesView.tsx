import * as React from "react";
import { useSelector } from "react-redux";
import { ApplicationState } from "../../../Store/Configuration";
import { Animated, Icon } from "../../../Shared/Components";

interface TechnologiesViewProps {
    className?: string;
}

export const TechnologiesView = (props: TechnologiesViewProps): React.ReactElement => {
    const data = useSelector((state: ApplicationState) => state.contentPageData);
    const technology = data?.components?.sectionTechnologies;

    return (
        <section className={`section ${props.className ?? ""}`}>
            <div className="bulma-container">
                <div className="py-6">
                    <Animated dataAos="fade-down">
                        <p className="is-size-3	has-text-centered has-text-link">
                            {technology?.caption?.toUpperCase()}
                        </p>
                    </Animated>
                    <div className="p-6">
                        <div className="bulma-columns ">
                            <div className="bulma-column">
                                <Animated dataAos="fade-up" className="is-flex is-align-items-center py-2">
                                    <Icon name="CodeBraces" size={1.2} className="has-text-link is-flex is-align-self-center mr-2" />
                                    <h3 className="is-size-4 py-5 has-text-black">
                                        {technology?.title1}
                                    </h3>
                                </Animated>
                                <Animated dataAos="fade-up">
                                    <h4 className="is-size-5 py-2 has-text-grey line-height-18">{technology?.text1}</h4>
                                </Animated>
                            </div>
                            <div className="bulma-column">
                                <Animated dataAos="fade-up" className="is-flex is-align-items-center py-2">
                                    <Icon name="Bookshelf" size={1.2} className="has-text-link is-flex is-align-self-center mr-2" />
                                    <h3 className="is-size-4 py-5 has-text-black">
                                        {technology?.title2}
                                    </h3>
                                </Animated>
                                <Animated dataAos="fade-up">
                                    <h4 className="is-size-5 py-2 has-text-grey line-height-18">{technology?.text2}</h4>
                                </Animated>
                            </div>
                        </div>
                        <div className="bulma-columns">
                            <div className="bulma-column">
                                <Animated dataAos="fade-up" className="is-flex is-align-items-center py-2">
                                    <Icon name="Server" size={1.2} className="has-text-link is-flex is-align-self-center mr-2" />
                                    <h3 className="is-size-4 py-5 has-text-black">
                                        {technology?.title3}
                                    </h3>
                                </Animated>
                                <Animated dataAos="fade-up">
                                    <h4 className="is-size-5 py-2 has-text-grey line-height-18">{technology?.text3}</h4>
                                </Animated>
                            </div>
                            <div className="bulma-column">
                                <Animated dataAos="fade-up" className="is-flex is-align-items-center py-2">
                                    <Icon name="Cloud" size={1.2} className="has-text-link is-flex is-align-self-center mr-2" />
                                    <h3 className="is-size-4 py-5 has-text-black">
                                        {technology?.title4}
                                    </h3>
                                </Animated>
                                <Animated dataAos="fade-up">
                                    <h4 className="is-size-5 py-2 has-text-grey line-height-18">{technology?.text4}</h4>
                                </Animated>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    );
};
