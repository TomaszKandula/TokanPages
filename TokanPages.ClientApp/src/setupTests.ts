// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
