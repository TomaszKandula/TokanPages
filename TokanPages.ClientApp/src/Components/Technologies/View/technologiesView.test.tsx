import "../../../setupTests";
import React from "react";
import * as Redux from "react-redux";
import { render } from "enzyme";
import { TechnologiesView } from "./technologiesView";

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn()
}));

describe("test component: technologiesView", () => {
    const testContent = {
        language: "eng",
        caption: "Technologies",
        header: "I work primarily with",
        title1: "Languages",
        text1: "Today in my daily job, I use...",
        title2: "Frameworks/Libraries",
        text2: "Back-End: NET Framework 4.5 (for one project)...",
        title3: "OR/M",
        text3: "I have started with Entity Framework...",
        title4: "Cloud Services",
        text4: "I have experience with...",
    };

    beforeEach(() => {
        jest.spyOn(Redux, "useSelector").mockReturnValueOnce({
            isLoading: false,
            content: testContent
        });
    });

    it("should render correctly '<TechnologiesView />' when content is loaded.", () => {
        const html = render(<TechnologiesView />);
        expect(html).toMatchSnapshot();
    });
});
