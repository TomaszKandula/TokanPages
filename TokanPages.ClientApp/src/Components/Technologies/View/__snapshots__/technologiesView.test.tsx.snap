// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: technologiesView should render correctly '<TechnologiesView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container"
        >
          <div
            class="py-6"
          >
            <div
              data-aos="fade-down"
              data-aos-delay="0"
            >
              <p
                class="is-size-3	has-text-centered has-text-link"
              >
                TECHNOLOGIES
              </p>
            </div>
            <div
              class="p-6"
            >
              <div
                class="bulma-columns"
              >
                <div
                  class="bulma-column"
                >
                  <div
                    class="is-flex is-align-items-center py-2"
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                        role="presentation"
                        style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8,3A2,2 0 0,0 6,5V9A2,2 0 0,1 4,11H3V13H4A2,2 0 0,1 6,15V19A2,2 0 0,0 8,21H10V19H8V14A2,2 0 0,0 6,12A2,2 0 0,0 8,10V5H10V3M16,3A2,2 0 0,1 18,5V9A2,2 0 0,0 20,11H21V13H20A2,2 0 0,0 18,15V19A2,2 0 0,1 16,21H14V19H16V14A2,2 0 0,1 18,12A2,2 0 0,1 16,10V5H14V3H16Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                    <h3
                      class="is-size-4 py-5 has-text-black"
                    >
                      Languages
                    </h3>
                  </div>
                  <div
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <h4
                      class="is-size-5 py-2 has-text-grey line-height-18"
                    >
                      Today in my daily job, I use...
                    </h4>
                  </div>
                </div>
                <div
                  class="bulma-column"
                >
                  <div
                    class="is-flex is-align-items-center py-2"
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                        role="presentation"
                        style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 3V18H12V3H9M12 5L16 18L19 17L15 4L12 5M5 5V18H8V5H5M3 19V21H21V19H3Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                    <h3
                      class="is-size-4 py-5 has-text-black"
                    >
                      Frameworks/Libraries
                    </h3>
                  </div>
                  <div
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <h4
                      class="is-size-5 py-2 has-text-grey line-height-18"
                    >
                      Back-End: NET Framework 4.5 (for one project)...
                    </h4>
                  </div>
                </div>
              </div>
              <div
                class="bulma-columns"
              >
                <div
                  class="bulma-column"
                >
                  <div
                    class="is-flex is-align-items-center py-2"
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                        role="presentation"
                        style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4,1H20A1,1 0 0,1 21,2V6A1,1 0 0,1 20,7H4A1,1 0 0,1 3,6V2A1,1 0 0,1 4,1M4,9H20A1,1 0 0,1 21,10V14A1,1 0 0,1 20,15H4A1,1 0 0,1 3,14V10A1,1 0 0,1 4,9M4,17H20A1,1 0 0,1 21,18V22A1,1 0 0,1 20,23H4A1,1 0 0,1 3,22V18A1,1 0 0,1 4,17M9,5H10V3H9V5M9,13H10V11H9V13M9,21H10V19H9V21M5,3V5H7V3H5M5,11V13H7V11H5M5,19V21H7V19H5Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                    <h3
                      class="is-size-4 py-5 has-text-black"
                    >
                      OR/M
                    </h3>
                  </div>
                  <div
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <h4
                      class="is-size-5 py-2 has-text-grey line-height-18"
                    >
                      I have started with Entity Framework...
                    </h4>
                  </div>
                </div>
                <div
                  class="bulma-column"
                >
                  <div
                    class="is-flex is-align-items-center py-2"
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                        role="presentation"
                        style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6.5 20Q4.22 20 2.61 18.43 1 16.85 1 14.58 1 12.63 2.17 11.1 3.35 9.57 5.25 9.15 5.88 6.85 7.75 5.43 9.63 4 12 4 14.93 4 16.96 6.04 19 8.07 19 11 20.73 11.2 21.86 12.5 23 13.78 23 15.5 23 17.38 21.69 18.69 20.38 20 18.5 20Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                    <h3
                      class="is-size-4 py-5 has-text-black"
                    >
                      Cloud Services
                    </h3>
                  </div>
                  <div
                    data-aos="fade-up"
                    data-aos-delay="0"
                  >
                    <h4
                      class="is-size-5 py-2 has-text-grey line-height-18"
                    >
                      I have experience with...
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container"
      >
        <div
          class="py-6"
        >
          <div
            data-aos="fade-down"
            data-aos-delay="0"
          >
            <p
              class="is-size-3	has-text-centered has-text-link"
            >
              TECHNOLOGIES
            </p>
          </div>
          <div
            class="p-6"
          >
            <div
              class="bulma-columns"
            >
              <div
                class="bulma-column"
              >
                <div
                  class="is-flex is-align-items-center py-2"
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                      role="presentation"
                      style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8,3A2,2 0 0,0 6,5V9A2,2 0 0,1 4,11H3V13H4A2,2 0 0,1 6,15V19A2,2 0 0,0 8,21H10V19H8V14A2,2 0 0,0 6,12A2,2 0 0,0 8,10V5H10V3M16,3A2,2 0 0,1 18,5V9A2,2 0 0,0 20,11H21V13H20A2,2 0 0,0 18,15V19A2,2 0 0,1 16,21H14V19H16V14A2,2 0 0,1 18,12A2,2 0 0,1 16,10V5H14V3H16Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                  <h3
                    class="is-size-4 py-5 has-text-black"
                  >
                    Languages
                  </h3>
                </div>
                <div
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <h4
                    class="is-size-5 py-2 has-text-grey line-height-18"
                  >
                    Today in my daily job, I use...
                  </h4>
                </div>
              </div>
              <div
                class="bulma-column"
              >
                <div
                  class="is-flex is-align-items-center py-2"
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                      role="presentation"
                      style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 3V18H12V3H9M12 5L16 18L19 17L15 4L12 5M5 5V18H8V5H5M3 19V21H21V19H3Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                  <h3
                    class="is-size-4 py-5 has-text-black"
                  >
                    Frameworks/Libraries
                  </h3>
                </div>
                <div
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <h4
                    class="is-size-5 py-2 has-text-grey line-height-18"
                  >
                    Back-End: NET Framework 4.5 (for one project)...
                  </h4>
                </div>
              </div>
            </div>
            <div
              class="bulma-columns"
            >
              <div
                class="bulma-column"
              >
                <div
                  class="is-flex is-align-items-center py-2"
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                      role="presentation"
                      style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4,1H20A1,1 0 0,1 21,2V6A1,1 0 0,1 20,7H4A1,1 0 0,1 3,6V2A1,1 0 0,1 4,1M4,9H20A1,1 0 0,1 21,10V14A1,1 0 0,1 20,15H4A1,1 0 0,1 3,14V10A1,1 0 0,1 4,9M4,17H20A1,1 0 0,1 21,18V22A1,1 0 0,1 20,23H4A1,1 0 0,1 3,22V18A1,1 0 0,1 4,17M9,5H10V3H9V5M9,13H10V11H9V13M9,21H10V19H9V21M5,3V5H7V3H5M5,11V13H7V11H5M5,19V21H7V19H5Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                  <h3
                    class="is-size-4 py-5 has-text-black"
                  >
                    OR/M
                  </h3>
                </div>
                <div
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <h4
                    class="is-size-5 py-2 has-text-grey line-height-18"
                  >
                    I have started with Entity Framework...
                  </h4>
                </div>
              </div>
              <div
                class="bulma-column"
              >
                <div
                  class="is-flex is-align-items-center py-2"
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center has-text-link is-flex is-align-self-center mr-2"
                      role="presentation"
                      style="width: 1.7999999999999998rem; height: 1.7999999999999998rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6.5 20Q4.22 20 2.61 18.43 1 16.85 1 14.58 1 12.63 2.17 11.1 3.35 9.57 5.25 9.15 5.88 6.85 7.75 5.43 9.63 4 12 4 14.93 4 16.96 6.04 19 8.07 19 11 20.73 11.2 21.86 12.5 23 13.78 23 15.5 23 17.38 21.69 18.69 20.38 20 18.5 20Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                  <h3
                    class="is-size-4 py-5 has-text-black"
                  >
                    Cloud Services
                  </h3>
                </div>
                <div
                  data-aos="fade-up"
                  data-aos-delay="0"
                >
                  <h4
                    class="is-size-5 py-2 has-text-grey line-height-18"
                  >
                    I have experience with...
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
